pandas/core/groupby/groupby.py

```diff
@@ -847,12 +847,12 @@ def f(g):
                 # on a string grouper column
 
                 with _group_selection_context(self):
-                    return self._python_apply_general(f)
+                    return self._python_apply_general(f, self._selected_obj)
 
         return result
 
-    def _python_apply_general(self, f):
-        keys, values, mutated = self.grouper.apply(f, self._selected_obj, self.axis)
+    def _python_apply_general(self, f, obj):
+        keys, values, mutated = self.grouper.apply(f, obj, self.axis)
```

Comment:
can u add a doc string and types here (at least for the added args)

Reply:
While adding the docstring, I realized that the name "obj" here is very confusing. I renamed to data, this is the name used in grouper.apply.